rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -64,10 +67,10 @@ struct FromGenericMetricsMessage {
 enum MetricValue {
     #[serde(rename = "c")]
     Counter(f64),
-    #[serde(rename = "s")]
-    Set(Vec<u64>),
-    #[serde(rename = "d")]
-    Distribution(Vec<f64>),
+    #[serde(rename = "s", deserialize_with = "encoded_series_deserializer")]
```

Comment:
I think it would be good if the deserializer fn name expressed that it handles the compatible "legacy" format since `EncodedSeries` already implements `Deserialize`.

Maybe something along the lines of `encoded_series_legacy_compat_deserializer` (I am really bad at naming things)

Reply:
How do you feel about just `compat_deserializer`, as the reader could infer legacy and new from `compat`?